tests/src/Base/BoundBox.cpp

```diff
@@ -0,0 +1,262 @@
+    Base::BoundBox3d copy(box);
+    EXPECT_EQ(copy.IsValid(), true);
+    EXPECT_EQ(copy.MinX, 0);
+    EXPECT_EQ(copy.MinY, 0);
+    EXPECT_EQ(copy.MinZ, 0);
+    EXPECT_EQ(copy.MaxX, 1);
+    EXPECT_EQ(copy.MaxY, 1);
+    EXPECT_EQ(copy.MaxZ, 1);
+}
+
+TEST(BoundBox, TestAssign)
+{
+    Base::BoundBox3d box;
+    box.Add(Base::Vector3d(0, 0, 0));
+    box.Add(Base::Vector3d(1, 1, 1));
+    Base::BoundBox3d copy;
+    copy = box;
+    EXPECT_EQ(copy.IsValid(), true);
+    EXPECT_EQ(copy.MinX, 0);
+    EXPECT_EQ(copy.MinY, 0);
+    EXPECT_EQ(copy.MinZ, 0);
+    EXPECT_EQ(copy.MaxX, 1);
+    EXPECT_EQ(copy.MaxY, 1);
+    EXPECT_EQ(copy.MaxZ, 1);
+}
+
+TEST(BoundBox, TestPoints)
+{
+    Base::Vector3d p[1];
+    Base::BoundBox3d box(p, 1);
+    EXPECT_EQ(box.IsValid(), true);
+    EXPECT_EQ(box.MinX, 0);
+    EXPECT_EQ(box.MinY, 0);
+    EXPECT_EQ(box.MinZ, 0);
+    EXPECT_EQ(box.MaxX, 0);
+    EXPECT_EQ(box.MaxY, 0);
+    EXPECT_EQ(box.MaxZ, 0);
+}
+
+TEST(BoundBox, TestDistance)
+{
+    Base::BoundBox3d box(Base::Vector3d(0, 0, 0), 0.5);
+    EXPECT_EQ(box.IsValid(), true);
+    EXPECT_EQ(box.MinX, -0.5);
+    EXPECT_EQ(box.MinY, -0.5);
+    EXPECT_EQ(box.MinZ, -0.5);
+    EXPECT_EQ(box.MaxX, 0.5);
+    EXPECT_EQ(box.MaxY, 0.5);
+    EXPECT_EQ(box.MaxZ, 0.5);
+}
+
+Base::BoundBox3d createBox(boost::beast::span<Base::Vector3d> span)
+{
+    return {span.data(), span.size()};
+}
+
+TEST(BoundBox, TestEmptySpan)
+{
+    Base::BoundBox3d box = createBox({});
+    EXPECT_EQ(box.IsValid(), false);
+}
+
+TEST(BoundBox, TestNonEmptySpan)
+{
+    std::vector<Base::Vector3d> points = {Base::Vector3d {1.0, 0.0, 0.0},
+                                          Base::Vector3d {0.0, 1.0, 0.0},
+                                          Base::Vector3d {0.0, 0.0, 1.0}};
+    Base::BoundBox3d box = createBox({points.data(), 2});
+    EXPECT_EQ(box.IsValid(), true);
+    EXPECT_EQ(box.MinX, 0);
+    EXPECT_EQ(box.MinY, 0);
+    EXPECT_EQ(box.MinZ, 0);
+    EXPECT_EQ(box.MaxX, 1);
+    EXPECT_EQ(box.MaxY, 1);
+    EXPECT_EQ(box.MaxZ, 0);
+}
+
+TEST(BoundBox, TestArray)
+{
+    std::vector<Base::Vector3d> points = {Base::Vector3d {1.0, 0.0, 0.0},
+                                          Base::Vector3d {0.0, 1.0, 0.0},
+                                          Base::Vector3d {0.0, 0.0, 1.0}};
+    Base::BoundBox3d box(points.data(), points.size());
+    EXPECT_EQ(box.IsValid(), true);
+    EXPECT_EQ(box.MinX, 0);
+    EXPECT_EQ(box.MinY, 0);
+    EXPECT_EQ(box.MinZ, 0);
+    EXPECT_EQ(box.MaxX, 1);
+    EXPECT_EQ(box.MaxY, 1);
+    EXPECT_EQ(box.MaxZ, 1);
+}
+
+TEST(BoundBox, TestIntersect)
+{
+    Base::BoundBox3d box1, box2, box3;
+    EXPECT_EQ(box1.Intersect(box1), false);
+    box1.Add(Base::Vector3d(0, 0, 0));
+    EXPECT_EQ(box1.Intersect(box1), true);
+    EXPECT_EQ(box1.Intersect(box2), false);
+    EXPECT_EQ(box2.Intersect(box1), false);
+
+    box1.Add(Base::Vector3d(1, 1, 1));
+    box2.Add(Base::Vector3d(0.5, 0.5, 0.5));
```

Comment:
I've been adding `// NOLINTBEGIN(readability-magic-numbers)` (and corresponding NOLINTEND) in the test files I work on, since tests often have these.